src/from_bytes_radix.rs

```diff
@@ -0,0 +1,376 @@
+/** A trait which provides `from_bytes_radix()` for integer types.
+
+This acts like `from_str_radix`, including panicking if `radix` is not in [2, 32].
+`0-9`, `A-Z` and `a-z` are supported as possible digits (case-insensitive.)
+However, there are a few minor differences to `from_str_radix` in the input and result types.
+```
+use bstr::{BStr, FromBytesRadix, IntErrorKind};
+
+for radix in 2..=36 {
+
+    let r = radix as u8;
+
+    let e = BStr::new(b"");
+    let empty = u8::from_bytes_radix(&e, 10);
+    assert_eq!(empty.unwrap_err().kind(), &IntErrorKind::Empty);
+
+    let a = BStr::new(b"11");
+    let eleven = u8::from_bytes_radix(&a, radix);
+    assert_eq!(eleven, Ok(r + 1));
+
+    let b = BStr::new("111111111");
+    let pos_overflow = u8::from_bytes_radix(&b, radix);
+    assert_eq!(pos_overflow.unwrap_err().kind(), &IntErrorKind::PosOverflow);
+
+    let c = BStr::new("-111");
+    let negatory = u8::from_bytes_radix(&c, radix);
+    assert_eq!(negatory.unwrap_err().kind(), &IntErrorKind::InvalidDigit);
+
+
+    let r = radix as i32;
+
+    let totally_fine = i32::from_bytes_radix(&c, radix);
+    assert_eq!(totally_fine, Ok(-(r*r + r + 1)));
+}
+```
+
+The `NonZero` versions of integers are not currently supported for parsing.
+Instead, please parse the equivalent possibly-zero integer, then convert:
+```
+use core::num::NonZeroU8;
+use bstr::{BStr, FromBytesRadix};
+
+let a = BStr::new(b"11");
+let eleven = u8::from_bytes_radix(&a, 10).ok().and_then(NonZeroU8::new);
+assert_eq!(eleven, NonZeroU8::new(11));
+
+let zero = BStr::new(b"0");
+let nada = u8::from_bytes_radix(&zero, 10).ok().and_then(NonZeroU8::new);
+assert_eq!(nada, None);
+```
+
+
+
+*/
+pub trait FromBytesRadix<T>
```

Comment:
Where is this type parameter `T` used? It looks like the associated type is what actually gets used?

But even then that associated type can be replaced with `Self`.

Reply:
Yeah @lopopolo you're 100% correct there. It's a residual bit from when I was trying things with generics rather than just resorting to a macro. Removed now.